packages/hierarchy-builder/src/hierarchy-builder/HierarchyProvider.ts

```diff
@@ -179,21 +201,56 @@ export class HierarchyProvider {
   }
 }
 
-/**
- * This is required because we don't have a way to convert a number to base36 through ECSQL. Need to
- * talk with ECSQL guys.
- */
-function convertECInstanceIdSuffixToBase36<TNode extends { label: string }>(node: TNode): TNode {
-  const m = node.label.match(/\s\[(0x[\w\d]+)\]$/);
-  const suffix = m?.at(1);
-  if (!suffix) {
-    return node;
-  }
+function preProcessNodes(hierarchyFactory: IHierarchyLevelDefinitionsFactory) {
+  return (nodes: Observable<HierarchyNode>): Observable<HierarchyNode> => {
+    return nodes.pipe(
+      concatMap(async (n) => (hierarchyFactory.preProcessNode ? hierarchyFactory.preProcessNode(n) : n)),
```

Comment:
I think it should be:
```suggestion
      concatMap((n) => (hierarchyFactory.preProcessNode ? from(hierarchyFactory.preProcessNode(n)) : of(n))),
```

Reply:
I don't think there's a difference, since `concatMap` accepts a mapping function returning a promise